pkg/cmd/pulumi/preview.go

```diff
@@ -417,6 +420,17 @@ func newPreviewCmd() *cobra.Command {
 				return result.FromError(err)
 			}
 
+			var defaultReferenceLoader schema.ReferenceLoader
+			if cwd, err := os.Getwd(); err == nil {
+				silentPluginContext, err := plugin.NewContext(nil, nil, nil, nil, cwd, nil, true, nil)
+				if err == nil {
+					// Initialize the reference loader for the happy path
+					// The engine can handle the case where the reference loader is nil, so we don't need to check for
+					// errors or hard fail here.
+					defaultReferenceLoader = schema.NewPluginLoader(silentPluginContext.Host)
+				}
+			}
+
```

Comment:
This shouldn't be setup here. Look at pkg/engine/deployment.go around line 180 where the opts.Host is used to setup the default host. That's the consistent place to also setup the loader as then you can use whichever host is already setup.

Reply:
Done ✅